src/Quaternion.jl

```diff
@@ -115,43 +115,115 @@ end
 
 argq(q::Quaternion) = normalizeq(Quaternion(0, q.v1, q.v2, q.v3))
 
-function exp(q::Quaternion)
+"""
+    extend_analytic(f, q::Quaternion)
+
+Evaluate the extension of the complex analytic function `f` to the quaternions at `q`.
+
+Given ``q = s + a u``, where ``s`` is the real part, ``u`` is a pure unit quaternion,
+and ``a \\ge 0`` is the magnitude of the imaginary part of ``q``,
+
+```math
+f(q) = \\Re(f(z)) + \\Im(f(z)) u,
+```
+is the extension of `f` to the quaternions, where ``z = a + s i`` is a complex analog to
+``q``.
+
+See Theorem 5 of [^Sudbery1970] for details.
+
+[^Sudbery1970]
+    Sudbery (1979). Quaternionic analysis. Mathematical Proceedings of the Cambridge 
+    Philosophical Society,85, pp 199­225
+    doi:[10.1017/S030500410005563](https://doi.org/10.1017/S0305004100055638)
+"""
+function extend_analytic(f, q::Quaternion)
+    a = abs_imag(q)
     s = q.s
-    se = exp(s)
-    scale = se
-    th = abs_imag(q)
-    if th > 0
-        scale *= sin(th) / th
+    z = complex(s, a)
+    w = f(z)
+    wr, wi = reim(w)
+    scale = wi / a
+    norm = _isexpfun(f) && iszero(s)
+    if a > 0
+        return Quaternion(wr, scale * q.v1, scale * q.v2, scale * q.v3, norm)
+    else  # quaternion may be real or complex
+        return Quaternion(wr, oftype(scale, wi), zero(scale), zero(scale), norm)
```

Comment:
Could you add more comments here? I think this corresponds to Pu(q)/|Pu(q)|, but it would be easier to understand if the special consideration of the first component is clearly stated.

Reply:
Is this clearer now?

Reply:
Thanks for the updates! I think it's clear enough.